### DISCUSSION TITLE
Suppress device selection dialog when requesting device

### DISCUSSION THREAD

**Discussion by joshberry** [inthehand/32feet#375] [Q&A] [answered]
I'm using the BluetoothLE package in a WPF application and would like to connect to a known BLE device without requiring the user to go through the dialog that is displayed when calling RequestDeviceAsync(). Is there a way to accomplish this?

Also, currently when calling RequestDeviceAsync() from my app, the dialog is not sized correctly (see below), making it impossible to interact with. I suspect this is some sort of bug caused by the fact that my app is minimized to the system tray or something because it works fine in a default wpf app template. If I can't bypass the dialog, I'm going to have to figure this one out I guess, so any ideas here would be appreciated as well.

_incorrectly size dialog_
<img width="251" alt="image" src="https://github.com/inthehand/32feet/assets/1396816/66965664-74d2-40b9-87a8-fdc95815098a">

Thanks!

**Comment by peterfoot** [comment, marked as answer]
Instead of RequestDeviceAsync you can use ScanForDevicesAsync to programmatically get a list of devices to connect to without any UI.

**Reply by joshberry** [reply]
Exactly what I needed. Thanks for the guidance!

Any ideas on why the selection dialog would appear like that? I may need to support device selection in the future.
